src/deploy/functions/runtimes/node/versioning.ts

```diff
@@ -34,30 +25,66 @@ export const FUNCTIONS_SDK_VERSION_TOO_OLD_WARNING =
   clc.bold("npm i --save firebase-functions@latest") +
   " in the functions folder.";
 
+/**
+ * Exported for testing purposes only.
+ *
+ * @internal
+ */
+export function findModuleVersion(name: string, resolvedPath: string): string | undefined {
+  let searchPath = path.dirname(resolvedPath);
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
```

Comment:
I think search path will always terminate with `/`, right? Why not use that as the condition instead of doing `while (true)`?